files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,122 @@
+# !/usr/bin/env python
+'''
+This code is for a specific daemon, process-docker.
+Which sends process and docker CPU/memory utilization data to DB every 2 mins.
+'''
+
+import sys
+import os
+import time
+import syslog
+import subprocess
+import re
+import swsssdk 
+from datetime import datetime
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "procdockerstats"
+
+REDIS_HOSTIP = "127.0.0.1"
+
+# ========================== Syslog wrappers ==========================
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+# ========================== ProcessDocker class ==========================
+class ProcDockerStats:
+
+    def __init__(self):
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB") 
+    
+    def run_command(self, cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+	(stdout, stderr) = proc.communicate()
+	
+	if proc.returncode != 0:
+	    log_error("Error running command '{}'".format(cmd))
+	else:
+            return stdout
+
+    def format_docker_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split("   +", lines[0])
+        x = len(lines)
+        dict1 = dict()
+        dict_list = []
+        for i in range(1,x):
```

Comment:
You should be able to simply state `for line in lines:`, then refer to `line` instead of `line[i]`. You could then also delete line 57

Reply:
I used range because I need to separate between keys and values. 1st row contains key names and 2nd row onward its values . Thoughts?

Reply:
I think you could ten iterate over the remainder of the lines using `for line in lines[1:]` (using a slice).